Add notes on passing --env to mx for the Eclipse setup and using MX_BUILD_EXPLODED/LINKY_LAYOUT (#2440)
oracle/truffleruby: doc/contributor/using-eclipse.md

* `TRUFFLERUBY-TEST.dist`
* `TRUFFLERUBY.dist`

> ### Running from the Eclipse Files Directly

Reviewer: FYI I think this no longer works (without what you added below)

Author: Moved that around to avoid confusion.

```bash
jt ruby -e 'p Truffle'
Truffle
```

## Advanced Setup

**Note:** This setup is not widely tested.

When building using specific env files, with `JT_ENV` or `--env`, mx will
need to be told about it explicitly. For instance:

```bash
jt mx --env jvm-ce eclipseinit
```

When working on the Graal/Truffle code itself, the following two lines will
configure the build so that incremental compilation and even hot code
replacement works:

```bash
MX_BUILD_EXPLODED=true

> LINKY_LAYOUT=*.jar

Reviewer: Should these two lines use `export`? Do you set them in some shell profile, or in a mx env file?

Author: I rephrased the paragraph before to explicitly point to the env file.

```

> These settings are not recommended for producing distributable artifacts, but

Reviewer: ```suggestion
These settings must not be used for producing distributable artifacts, but
```
Because it will actually just be a bunch of broken symlinks
